### PR TITLE
TST: Added regression test case

### LINKED ISSUE #18549: boolean frames multiplied by floats have dtypes=object
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
(pd.DataFrame(True, list('ab'), list('cd')) * 1.0).dtypes
#  returns object

type(True * 1.0)
# returns float
```
#### Problem description
A boolean frame multiplied by a float should return a DataFrame of floats to be consistent with python scalar operations (e.g., bool * float -> float).  Instead it returns dtypes = `object`

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-36-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.21.0
pytest: 3.2.5
pip: 9.0.1
setuptools: 37.0.0
Cython: 0.27.3
numpy: 1.13.3
scipy: 1.0.0
pyarrow: None
xarray: 0.10.0
IPython: 5.5.0
sphinx: None
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: 2.6.4
feather: None
matplotlib: 2.1.0
openpyxl: None
xlrd: None
xlwt: 1.3.0
xlsxwriter: None
lxml: None
bs4: None
html5lib: 1.0b10
sqlalchemy: 1.1.15
pymysql: None
psycopg2: 2.7.3.2 (dt dec pq3 ext lo64)
jinja2: 2.8.1
s3fs: None
fastparquet: None
pandas_gbq: 0.2.0
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by ShaharNaveh**
Resurrection of #28966

- [x] closes #18549
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/arithmetic/test_numeric.py [hunk 1]
```diff
@@ -1284,3 +1287,35 @@ def test_dataframe_div_silenced():
     )
     with tm.assert_produces_warning(None):
         pdf1.div(pdf2, fill_value=0)
+
+
+class TestNumericArraylikeArithmeticWithBool:
+    @pytest.mark.skipif(
+        compat.is_platform_32bit(),
```
Comment (WillAyd) [inline pandas/tests/arithmetic/test_numeric.py]: Hmm what’s this for? Should still work on 32 bit

File: pandas/tests/arithmetic/test_numeric.py [hunk 2]
```diff
@@ -1284,3 +1287,35 @@ def test_dataframe_div_silenced():
...
+            "mul",
+            "pow",
+            "rfloordiv",
+            "rpow",
+            "rmod",
+            "rmul",
+            "rtruediv",
+            "truediv",
+        ]:
+            pytest.skip("Arithmetic operation is not supported")
```
Comment (WillAyd) [inline pandas/tests/arithmetic/test_numeric.py]: Probably better to xfail than skip here; don’t think it matters when used imperatively but former would be more fitting

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
off-topic @MomIsBestFriend if you're up for suggestions on things needing attention, in core.indexing L1983 is never reached in the tests and i _think_ it may be impossible, could use a fresh set of eyes to see if we can prove that hypothesis.  Also L892-901 is not reached, but i dont have a strong intuition for if that is possible.

File: pandas/tests/arithmetic/test_numeric.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_numeric.py]: is this something that _should_ be supported but isnt?

**Comment by MomIsBestFriend** [comment]
@jbrockmendel Sure!

I always like it when you give me tasks:)
I don't have a lot of knowledge, but I am willing to learn.

Thank you, means a lot :+1:

File: pandas/tests/arithmetic/test_numeric.py (hunk 2 above)
Comment (MomIsBestFriend) [reply pandas/tests/arithmetic/test_numeric.py]: > is this something that _should_ be supported but isnt?

I am not sure.

File: pandas/tests/arithmetic/test_numeric.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/tests/arithmetic/test_numeric.py]: @WillAyd Can you please look at the reasons b38f9e6 failed.

That's what I am understanding, from looking at it.

Maybe you have other conclusions.

Comment (WillAyd) [reply pandas/tests/arithmetic/test_numeric.py]: I think its just a bug somewhere; we typically don't default back to platform ints

Comment (MomIsBestFriend) [reply pandas/tests/arithmetic/test_numeric.py]: @WillAyd Should I put a ```pytest.mark.skipif``` decorator (skiping this test, if running on 32-bit platform) at least for now?

**Comment by jbrockmendel** [comment]
re-started azure

File: pandas/tests/arithmetic/test_numeric.py [hunk 3]
```diff
@@ -1306,3 +1306,31 @@ def test_dataframe_div_silenced():
...
+            "mul",
+            "pow",
+            "rfloordiv",
+            "rpow",
+            "rmod",
+            "rmul",
+            "rtruediv",
+            "truediv",
+        ]:
+            pytest.xfail("Arithmetic operation is not supported")
```
Comment (jbrockmendel) [inline pandas/tests/arithmetic/test_numeric.py]: is this not supported as in "not meaningful and so never will be supported" or as in "we havent gotten around to it yet so the behavior is wrong"?  only the latter should be xfail

Comment (MomIsBestFriend) [reply pandas/tests/arithmetic/test_numeric.py]: I am not sure, maybe @jorisvandenbossche knows?

Comment (jorisvandenbossche) [reply pandas/tests/arithmetic/test_numeric.py]: It's hard to see what the test is actually testing, but I would think something like this?

```
In [5]: pd.Series([True]) * 3 
Out[5]: 
0    3
dtype: int64
```

but that doesn't fail, though.

But I agree with @jbrockmendel: if there are cases that it raises an error purposefully, we should test that and not xfail.

Comment (MomIsBestFriend) [reply pandas/tests/arithmetic/test_numeric.py]: I am not able to test this without having ```if/else``` statement in the code, which is not really a pattern we want in our tests (I think), Any thought on how to write this test case?

File: pandas/tests/arithmetic/test_numeric.py [hunk 4]
```diff
@@ -1306,3 +1306,31 @@ def test_dataframe_div_silenced():
     )
     with tm.assert_produces_warning(None):
         pdf1.div(pdf2, fill_value=0)
+
+
+class TestNumericArraylikeArithmeticWithBool:
+    @pytest.mark.parametrize("num", [1, 1.0])
```
Comment (jbrockmendel) [inline pandas/tests/arithmetic/test_numeric.py]: complex(1), np.int64(1), ... We may even have a fixture just called "one"

File: pandas/tests/arithmetic/test_numeric.py [hunk 5]
```diff
@@ -1306,3 +1306,31 @@ def test_dataframe_div_silenced():
...
+            "rmul",
+            "rtruediv",
+            "truediv",
+        ]:
+            pytest.xfail("Arithmetic operation is not supported")
+        expected = [op(num, num)]
+        expected = tm.box_expected(expected, box_with_array)
+        bool_box = tm.box_expected([True], box_with_array)
+        tm.assert_equal(expected, op(bool_box, num))
+        tm.assert_equal(expected, op(num, bool_box))
```
Comment (jbrockmendel) [inline pandas/tests/arithmetic/test_numeric.py]: nitpick, the pattern we usually use is:
```
bool_box = [True]
expected = [op(num, num)]

bool_box = tm.box_expected(...
expected = tm.box_expected(...

result = op(bool_box, num)
tm.assert_equal(...
result = op(num, bool_box)
tm.assert_equal(...
```

File: pandas/tests/arithmetic/test_numeric.py (hunk 3 above)
Comment (jorisvandenbossche) [reply pandas/tests/arithmetic/test_numeric.py]: The if/else might be fine, but in this if block, you can do a pytest.raises instead of xfail?

Comment (MomIsBestFriend) [reply pandas/tests/arithmetic/test_numeric.py]: Trying to test the actual errors, will result in having a test case that looks somewhat like this:

```python
   @pytest.mark.parametrize(
        "op",
        [
            operator.mul,
            ops.rmul,
            operator.floordiv,
            ops.rfloordiv,
            operator.truediv,
            ops.rtruediv,
            operator.pow,
            ops.rpow,
            operator.mod,
            ops.rmod,
        ],
    )
    @pytest.mark.parametrize("num", [np.int64(1), 1, 1.0])
    def test_array_like_bool_and_num_op_coerce_raises(self, op, num, box_with_array):
        # https://github.com/pandas-dev/pandas/issues/18549
        bool_box = [True]

        if op.__name__ in {"foo", "bar", "baz"}:
            msg = "|".join(
                [
                    "can't mod complex numbers",
                    "can't take floor of complex number",
                    r"unsupported operand type\(s\) for %: 'float' and 'Index'",
                    "cannot perform __foo__ with this index type: Index",
                ]
            )

            with pytest.raises(TypeError, match=msg):
                expected = [op(num, num)]
            return
        else:
            expected = [op(num, num)]

        bool_box = tm.box_expected(bool_box, box_with_array)
        expected = tm.box_expected(expected, box_with_array)

        if op.__name__ in {"foo", "bar", "baz"}:
            msg = "|".join(["cannot perform __foo__ with this index type: Index"])
            with pytest.raises(TypeError, match=msg):
                tm.assert_equal(expected, op(bool_box, num))
        else:
            tm.assert_equal(expected, op(bool_box, num))
        tm.assert_equal(expected, op(num, bool_box))
```

This looks sloppy, and not really readable, any thoughts?

Comment (jorisvandenbossche) [reply pandas/tests/arithmetic/test_numeric.py]: That looks fine to me in general (have some comments on some details, but that will be easier to add when this is in the diff)

**Comment by MomIsBestFriend** [comment]
Closing this, as this PR is giving me more headache than I expected.

**Pull request closed without merging**